core/src/main/java/org/infinispan/commands/CommandUUID.java

```diff
@@ -0,0 +1,94 @@
+package org.infinispan.commands;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.marshall.core.Ids;
+import org.infinispan.remoting.transport.Address;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Represents an unique identified for non-transaction write commands.
+ *
+ * It is used to lock the key for a specific command.
+ *
+ * @author Pedro Ruivo
+ * @since 8.0
+ */
+public class CommandUUID {
```

Comment:
The `UUID` name usually implies a particular representation: http://tools.ietf.org/html/rfc4122.html

`CommandId` would also be too close to our existing command ids, so I would suggest something like `CommandInvocationId` or `NonTxInvocationId`.